addons/marketing_campaign/models/marketing_campaign_segment.py

```diff
@@ -0,0 +1,144 @@
+    object_id = fields.Many2one(
+        related='campaign_id.object_id', string='Resource')
+    ir_filter_id = fields.Many2one('ir.filters', string='Filter', ondelete="restrict",
+                                   help="Filter to select the matching resource records that belong to this segment. "
+                                   "New filters can be created and saved using the advanced search on the list view of the Resource. "
+                                   "If no filter is set, all records are selected without filtering. "
+                                   "The synchronization mode may also add a criterion to the filter.")
+    sync_last_date = fields.Datetime(
+        string='Last Synchronization', help="Date on which this segment was synchronized last time (automatically or manually)")
+    sync_mode = fields.Selection([('create_date', 'Only records created after last sync'),
+                                  ('write_date',
+                                   'Only records modified after last sync (no duplicates)'),
+                                  ('all', 'All records (no duplicates)')],
+                                 string='Synchronization mode', default='create_date', help="Determines an additional criterion to add to the filter when selecting new records to inject in the campaign. "
+                                 '"No duplicates" prevents selecting records which have already entered the campaign previously.'
+                                 'If the campaign has a "unique field" set, "no duplicates" will also prevent selecting records which have '
+                                 'the same value for the unique field as other records that already entered the campaign.')
+    state = fields.Selection([('draft', 'New'),
+                              ('cancelled', 'Cancelled'),
+                              ('running', 'Running'),
+                              ('done', 'Done')],
+                             'Status', copy=False, default='draft')
+    date_run = fields.Datetime(
+        string='Launch Date', help="Initial start date of this segment.")
+    date_done = fields.Datetime(
+        string='End Date', help="Date this segment was last closed or cancelled.")
+    date_next_sync = fields.Datetime(compute='_compute_next_sync', string='Next Synchronization',
+                                     help="Next time the synchronization job is scheduled to run automatically")
+
+    def _compute_next_sync(self):
+        # next auto sync date is same for all segments
+        sync_job = self.env.ref(
+            'marketing_campaign.ir_cron_marketing_campaign_every_day')
+        self.date_next_sync = sync_job.nextcall
+
+    @api.constrains('ir_filter_id', 'campaign_id')
+    def _check_model(self):
+        for obj in self:
+            if not self.ir_filter_id:
+                return True
+            if self.campaign_id.object_id.model != self.ir_filter_id.model_id:
+                raise ValueError(
```

Comment:
ValidationError looks more appropriate for pythonic constraints, even if you raise ValueError _validate_fields eventually converts it to ValidationError